www/vanv08/hw08/SP2/assets/js/script.js

```diff
@@ -0,0 +1,296 @@
+    let body = "grant_type=refresh_token";
+    body += "&refresh_token=" + refresh_token;
+    body += "&client_id=" + client_id;
+    callAuthorizationApi(body);
+}
+// api call to get acces token
+function callAuthorizationApi(body) {
+    let xhr = new XMLHttpRequest();
+    xhr.open("POST", TOKEN, true);
+    xhr.setRequestHeader('Content-Type', 'application/x-www-form-urlencoded');
+    xhr.setRequestHeader('Authorization', 'Basic ' + btoa(client_id + ":" + client_secret));
+    xhr.send(body);
+    xhr.onload = handleAuthorizationResponse;
+}
+
+
+// Understanding response from server
+function handleAuthorizationResponse() {
+    if (this.status == 200) {
+        var data = JSON.parse(this.responseText);
+        console.log(data);
+        var data = JSON.parse(this.responseText);
+        if (data.access_token != undefined) {
+            access_token = data.access_token;
+            localStorage.setItem("access_token", access_token);
+        }
+        if (data.refresh_token != undefined) {
+            refresh_token = data.refresh_token;
+            localStorage.setItem("refresh_token", refresh_token);
+        }
+        onPageLoad();
+    } else {
+        console.log(this.responseText);
+
+    }
+}
+
+function refreshDevices() {
+    callApi("GET", DEVICES, null, handleDevicesResponse);
+}
+
+function handleDevicesResponse() {
+    if (this.status == 200) {
+        var data = JSON.parse(this.responseText);
+        console.log(data);
+        removeAllItems("devices");
+        data.devices.forEach(item => addDevice(item));
+    } else if (this.status == 401) {
+        refreshAccessToken()
+    } else {
+        console.log(this.responseText);
+
+    }
+}
+
+function addDevice(item) {
+    let node = document.createElement("option");
+    node.value = item.id;
+    node.innerHTML = item.name;
+    document.getElementById("devices").appendChild(node);
+}
+
+function callApi(method, url, body, callback) {
+    let xhr = new XMLHttpRequest();
+    xhr.open(method, url, true);
+    xhr.setRequestHeader('Content-Type', 'application/json');
+    xhr.setRequestHeader('Authorization', 'Bearer ' + access_token);
+    xhr.send(body);
+    xhr.onload = callback;
+}
+
+function refreshPlaylists() {
+    callApi("GET", PLAYLISTS, null, handlePlaylistsResponse);
+}
+
+function handlePlaylistsResponse() {
+    if (this.status == 200) {
+        var data = JSON.parse(this.responseText);
+        console.log(data);
+        removeAllItems("playlists");
+        data.items.forEach(item => addPlaylist(item));
+        document.getElementById('playlists').value = currentPlaylist;
+    } else if (this.status == 401) {
+        refreshAccessToken()
+    } else {
+        console.log(this.responseText);
+
+    }
+}
+
+function addPlaylist(item) {
+    let node = document.createElement("option");
+    node.value = item.id;
+    node.innerHTML = item.name + " (" + item.tracks.total + ")";
+    document.getElementById("playlists").appendChild(node);
```

Comment:
vybrat predem

Reply:
nejsem si jist zda chápu, co chcete bych upravil

Reply:
kolikrat v kodu volate ` document.getElementById("playlists")` ?

Reply:
Přesně 3x

Reply:
Ah mám to chápat, že je lepší si "playlist" uložit do var? A poté se na ni odkazovat, namísto abych volal document.getElementID?

Reply:
exactly